src/Services/PatientService.php

```diff
@@ -27,64 +29,16 @@ class PatientService extends BaseService
      */
     private $patient_picture_fallback_id = -1;
 
-    private $validator;
+    private $patientValidator;
 
     /**
      * Default constructor.
      */
     public function __construct()
     {
         parent::__construct('patient_data');
+        $this->patientValidator = new PatientValidator();
     }
-
-    // make this a comprehensive validation
-    public function validate($patient, $context, $id = null)
```

Comment:
The validation logic has been extracted out into the BaseValidator and PatientValidator classes/components.